include/swift/AST/DiagnosticsSema.def

```diff
@@ -3546,18 +3546,14 @@ ERROR(pattern_type_mismatch_context,none,
 
 ERROR(tuple_pattern_in_non_tuple_context,none,
       "tuple pattern cannot match values of the non-tuple type %0", (Type))
-WARNING(matching_pattern_with_many_assoc_values, none,
-        "cannot match several associated values at once, "
-        "implicitly tupling the associated values and trying to match that "
-        "instead", ())
-WARNING(matching_tuple_pattern_with_many_assoc_values,none,
-        "a tuple pattern cannot match several associated values at once, "
-        "implicitly tupling the associated values and trying to match "
-        "that instead", ())
-WARNING(matching_many_patterns_with_tupled_assoc_value,none,
-        "the enum case has a single tuple as an associated value, but "
-        "there are several patterns here, implicitly tupling the patterns "
-        "and trying to match that instead", ())
+WARNING(found_one_pattern_for_several_associated_values,none,
+        "enum case '%0' has %1 associated values; matching them as a tuple "
+        "is deprecated", (StringRef, unsigned))
```

Comment:
Apologies if I've paged something out, but why is this the only one of the three diagnostics which states that doing this is deprecated?

Reply:
In the other two cases, we most likely have a fix-it that either adds or removes parens (for example, you can look at the test cases), so I felt that providing an additional "this is deprecated" was not so useful. I can add that though, it is not a big deal.

Reply:
@brentdax what do you think of my response? Ok to not have the deprecated note because we are supplying fix-its? Or should I still add it?